src/main/antlr4/FIRRTL.g4

```diff
@@ -29,7 +29,16 @@ import firrtl.LexerHelper;
 
 // Does there have to be at least one module?
 circuit
-  : 'circuit' id ':' info? INDENT module* DEDENT EOF
+  : version? NEWLINE* 'circuit' id ':' info? INDENT module* DEDENT EOF
+  ;
+
+version
+  : 'FIRRTL' 'version' semver NEWLINE
+  ;
+
+// Due to lexer problems, something like 1.1.0 is lexed as DoubleLit '.' UnsignedInt
+semver
+  : SemVer .*?
```

Comment:
What is `.*?` intended to be matching? We should try to avoid non-greedy matching in general.

Reply:
That is intended to match any additional version information that could be included in the semantic version (e.g., for 1.2.0-rc1).

Reply:
Can you add a comment that mentions that? 

I suspect there are no performance concerns because I believe the `.` does not include newlines so in practice this won't lookahead very far, but can you do a sanity check that on a very large `.fir` file this change doesn't impact parsing speed?

Reply:
FWIW the firrtl spec doesn't allow for any such modifiers

Reply:
Okay, I see that in the grammar definition in the FIRRTL spec this is restricted to just X.Y.Z, so I'll remove this from here.